File: JavaScript/solveTriangle.js
function triangle(num) {
    let result = '';
    const increments = [0]
    for (let i = num - 1; i > 0; i--) {
        increments.push(i)
    }
    for (let i = 1; i < num + 1; i++){
        let prev = i;
        for (let j = 0; j < i; j++) {
            prev += increments[j]
            result += `${prev} `
        }
        result += '\n'
    }
    return result
}

console.log(triangle(5))

File: Python/solveTriangle.py
def row(num):
    result = ''
    increments = [i for i in range(num-1, 0, -1)]
    increments = [0] + increments
    for i in range(1, num+1):
        prev = i
        for j in range(i):
            prev = prev + increments[j]
            result += '{} '.format(prev)
        result += '\n'
    return result


print(row(9))
